File: library/src/main/java/com/pengrad/telegrambot/model/request/InputMediaVoiceNote.java
package com.pengrad.telegrambot.model.request;

import com.pengrad.telegrambot.request.ContentTypes;

import java.io.File;
import java.io.Serializable;

public class InputMediaVoiceNote extends InputMedia<InputMediaVoiceNote> implements Serializable {
    private final static long serialVersionUID = 0L;

    private Integer duration;

    public InputMediaVoiceNote(String media) {
        super("voice_note", media);
    }

    public InputMediaVoiceNote(File media) {
        super("voice_note", media);
    }

    public InputMediaVoiceNote(byte[] media) {
        super("voice_note", media);
    }

    public InputMediaVoiceNote duration(Integer duration) {
        this.duration = duration;
        return this;
    }

    @Override
    public String getDefaultFileName() {
        return ContentTypes.VOICE_FILE_NAME;
    }

    @Override
    public String getDefaultContentType() {
        return ContentTypes.VOICE_MIME_TYPE;
    }
}

File: library/src/main/java/com/pengrad/telegrambot/model/request/InputMediaVoiceNote.kt
package com.pengrad.telegrambot.model.request

import com.pengrad.telegrambot.request.ContentTypes
import java.io.File

class InputMediaVoiceNote : InputMedia<InputMediaVoiceNote> {

    constructor(media: String) : super("voice_note", media)
    constructor(media: File) : super("voice_note", media)
    constructor(media: ByteArray) : super("voice_note", media)

    private var duration: Int? = null

    fun duration(duration: Int): InputMediaVoiceNote {
        this.duration = duration
        return this
    }

    override fun getDefaultFileName(): String = ContentTypes.VOICE_FILE_NAME
    override fun getDefaultContentType(): String = ContentTypes.VOICE_MIME_TYPE
}
